### DISCUSSION TITLE
Why do we need "batch_idx" argument for "training_step" method?

### DISCUSSION THREAD

**Discussion by flourish-727** [Lightning-AI/pytorch-lightning#19707] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hello all. I am a rookie and just started learning Lightning. 
I don't really understand the role of "batch_idx" as an argument of the methods like "training_step" or validation_step. 
The method does not really use the argument (see this example from official):

**def training_step(self, batch, batch_idx):
    x, y, z = batch
    out = self.encoder(x)
    loss = self.loss(out, x)
    return loss**

See? "batch_idx" goes nowhere!!!
Could someone please give some hints?
Thanks!

**Comment by CA4GitHub** [comment]
Did you get an answer?  I'm curious too.
